nix/shared.nix

```diff
@@ -399,6 +399,11 @@ let
                           haskellPackagesNew.semigroups
                         ];
 
+                    managed =
+                      pkgsNew.haskell.lib.addBuildDepend
+                        haskellPackagesOld.managed
+                        haskellPackagesNew.semigroups;
```

Comment:
How is this change related to the PR?

Reply:
It's not.  It's a small change I made to get `nix-shell` working for GHC 7.10.3 when debugging a build failure